client/authority-discovery/src/error.rs

```diff
@@ -59,11 +57,11 @@ pub enum Error {
 	#[error("Failed to parse a libp2p key.")]
 	ParsingLibp2pIdentity(#[from] libp2p::identity::error::DecodingError),
 
-	#[error("Failed to sign using a specific public key.")]
-	MissingSignature(CryptoTypePublicPair),
+	#[error("Failed to sign network message. Reason: {0}.")]
+	CannotSignNetwork(String),
 
-	#[error("Failed to sign using all public keys.")]
-	Signing,
+	#[error("Failed to sign using key: {0:?}. Reason: {1}.")]
```

Comment:
1. It'd be better to print the key as hex; `{0:?}` will print it as e.g. `[0, 170, 255, 18, 35, 52]` which is not that nice to read when the key gets too long. (You could probably use `sp_core::hexdisplay::HexDisplay`.)
2. I'd probably put the key *after* the error message (which I think would be nicer in case the key is long), e.g. maybe something like this: `Failed to sign: {} (key = {})`, but it's up to you.